example/spec/parallel/ingest/ingestFromPdrWithChildWorkflowMetaSpec.js

```diff
@@ -0,0 +1,181 @@
+      testSuffix = createTestSuffix(testId);
+      testDataFolder = createTestDataPath(testId);
+
+      pdrFilename = `${testSuffix.slice(1)}_${origPdrFilename}`;
+
+      provider = { id: `s3_provider${testSuffix}` };
+
+      // populate collections, providers and test data
+      const populatePromises = await Promise.all([
+        updateAndUploadTestDataToBucket(
+          config.bucket,
+          s3data,
+          testDataFolder,
+          [
+            { old: 'cumulus-test-data/pdrs', new: testDataFolder },
+            { old: 'DATA_TYPE = MOD09GQ;', new: `DATA_TYPE = MOD09GQ${testSuffix};` },
+          ]
+        ),
+        uploadTestDataToBucket(
+          config.bucket,
+          unmodifiedS3Data,
+          testDataFolder
+        ),
+        addCollections(config.stackName, config.bucket, collectionsDir, testSuffix, testId),
+        addProviders(config.stackName, config.bucket, providersDir, config.bucket, testSuffix),
+      ]);
+
+      addedCollection = populatePromises[2][0];
+
+      // Rename the PDR to avoid race conditions
+      await s3().copyObject({
+        Bucket: config.bucket,
+        CopySource: `${config.bucket}/${testDataFolder}/${origPdrFilename}`,
+        Key: `${testDataFolder}/${pdrFilename}`,
+      }).promise();
+
+      await deleteS3Object(config.bucket, `${testDataFolder}/${origPdrFilename}`);
+
+      executionNamePrefix = randomString(3);
+
+      workflowExecution = await buildAndExecuteWorkflow(
+        config.stackName,
+        config.bucket,
+        workflowName,
+        { name: addedCollection.name, version: addedCollection.version },
+        provider,
+        undefined,
+        {
+          provider_path: testDataFolder,
+          executionNamePrefix,
+        }
+      );
+
+      queuePdrsOutput = await lambdaStep.getStepOutput(
+        workflowExecution.executionArn,
+        'QueuePdrs'
+      );
+    } catch (error) {
+      beforeAllFailed = true;
+      throw error;
+    }
+  });
+
+  afterAll(async () => {
+    // clean up stack state added by test
+    await Promise.all([
+      deleteFolder(config.bucket, testDataFolder),
+      cleanupCollections(config.stackName, config.bucket, collectionsDir, testSuffix),
+      cleanupProviders(config.stackName, config.bucket, providersDir, testSuffix),
+      apiTestUtils.deletePdr({
+        prefix: config.stackName,
+        pdr: pdrFilename,
+      }),
+    ]);
+  });
+
+  it('executes successfully', () => {
+    if (beforeAllFailed) fail('beforeAll() failed');
+    else {
+      expect(workflowExecution.status).toEqual('SUCCEEDED');
+    }
+  });
+
+  it('results in an IngestGranule workflow execution', async () => {
+    if (beforeAllFailed) fail('beforeAll() failed');
+    else {
+      const executionArn = queuePdrsOutput.payload.running[0];
+      await expectAsync(waitForStartedExecution(executionArn)).toBeResolved();
+    }
+  });
+
+  it('passes through childWorkflowMeta to the IngestGranule execution', async () => {
```

Comment:
Do we need the `beforeAll` check in all tests in the `describe` block?

`if (beforeAllFailed) fail('beforeAll() failed');`

I'm thinking it's on the previous two tests because these tests are run in parallel and any of them could be the first executed after `beforeAll`?

Reply:
Yes, the purpose is to *stop* the test from running if the beforeAll fails, so we need that clause in every test. Jasmine does *not* stop testing because test setup failed.

Reply:
Gotchya. So then we would need it in this test too?

Reply:
Fair point. Good catch. https://github.com/nasa/cumulus/pull/2114/commits/cd6bd67522c5a94620f8fee05ab67f2d4d8dca5e